### ISSUE TITLE
Remove "Sent from my iPhone"

### ISSUE THREAD

**Issue by thenktor** [deltachat/deltachat-android#118]
Hi,

messages sent from iOS devices have the following pseudo signatures:

- Sent from my iPhone
- Sent from my iPad
- Von meinem iPhone gesendet (German)
- Von meinem iPad gesendet (German)

Sadly Apple devices don't add a real signature but just add this as last line to the email. I think these can be removed/hidden.

**Comment by r10s** [comment]
Hm, plus a hundred other languages. I've also seen similar signatures from Samsung. If we start removing such things, we're in the uncomfortable situation to react to changes others do. However, I do not have a strong opinion to this currently, maybe it is pragmatic to just remove these strings.

**Comment by thenktor** [comment]
Yep, that's right. But I think removing the Apple strings may be a good idea because there is no iOS client and also those Apple devices probably have a greater user base than other manufacturers. And at least: maybe other manufacturers use real signatures.

**Comment by luisfsr** [comment]
What about giving the user the ability to set which phrases should be hidden? Perhaps a field or box, in advanced settings, or somewhere else, where signatures and other common sentences could be predefined by the user to be hidden by default of the conversation page...

**Comment by r10s** [comment]
And the editable input field comes with predefined values as ....

```
Sent from my iPhone
Sent from my iPad
```

... well, maybe?

**Comment by luisfsr** [comment]
That'd be really useful if we could use regular expressions. On my language (Portuguese) there are a lot of signatures and even email informations (as date of the previous messages that were sent by me) that are annoying and are not predefined to be hidden (but probably are hidden in English and a couple of most spoken languages). Maybe I (and other translators) could make a list of these expressions in Portuguese (and other respective languages) to help improve the list of expressions that should be hidden by default.

**Comment by r10s** [comment]
Maybe we have some other Ideas on this point, one question:

Where do these pseudo-footnotes typically get? Before or after a fullquote?

This **(A)**:

    My text
    
    Sent from my superDevice
    
    > Original message text

or that **(B)**:

    My text
    
    > Original message text

    Sent from my superDevice

?

**Comment by luisfsr** [comment]
Before **(A)**, both in cases of this issue, as well as issue #167.

**Comment by Ji-eF** [comment]
Could it be possible to encapsulate or start (each new line of a) Chat messages with a character other than -- or > ? For the next example, I choose =.
This would remove the need to filter signature and quotes completely... You just have to show new lines starting with =, stripping it if desired. If possible and 'legal', it could also be possible to use an invisible character...

````
On date, person wrote
> Quote
> Quote
> Quote

= Message body
= Message body
= Message body

--
δελτα Chat -- Instant chat with your e-mail address
````

**Comment by comradekingu** [comment]
A [hair space](https://en.wikipedia.org/wiki/Whitespace_character) or [zero-width space](https://en.wikipedia.org/wiki/Zero-width_space would) preserve the indentation.

**Comment by Two-Finger** [comment]
@comradekingu Or maybe even a sequence of them.
There are 6 distinct zero-width spaces in Unicode. ([Whitespace](https://en.wikipedia.org/wiki/Unicode_character_property#Whitespace), see the *second* table.)
A sequence of them would be like hidden code that's very unlikely to be a valid use. (Although a single ZWSP on the beginning of a line is invalid enough, I guess.)
And its total visible width will be zero.

**Comment by r10s** [comment]
> Could it be possible to encapsulate or start (each new line of a) Chat messages with a character other than -- or > ? For the next example, I choose =.

@Ji-eF Not sure if i got completely what you mean. You would like to prefix each line that is sent by Delta Chat  with a prefix similar to quotes, eg. `=`?

> This would remove the need to filter signature and quotes completely... [...]

I do not think that this would remove to filter signatures and quotes - the message may still come from a non-delta-chat client which adds footnotes, full quotes etc.

Apart from there are approaches to make Delta Chat messages look less strange in normal clients (eg. #136 or #125) - adding such a prefix would be a step in the wrong direction imho.

**Comment by Ji-eF** [comment]
> the message may still come from a non-delta-chat client which adds footnotes, full quotes etc.

You're right, I did not think about other MTA :/

**Comment by comradekingu** [comment]
@Two-Finger It is best to do this through a placeholder to do it, as sensible translation platforms (like Weblate) have built-in checks against these types of spaces, since they are most often used in error.

**Comment by testbird** [comment]
Isn't this solved by https://github.com/deltachat/deltachat-android/issues/112 ?
